### PR TITLE
Fix #8168 Add io.cache.{read,write} 

### LINKED ISSUE #8168: Add io.cache.read and io.cache.write
io.cache will set read and write to true or false when changing, but having io.cache.write enabled without read will catch all write operations and cache them but bypassing the call to the io backend. this way it is possible to create patchsets from a debugger session

### PR DESCRIPTION
**Pull request by MaskRay**
Based on https://github.com/radare/radare2/pull/8402

### CODE REVIEW

File: libr/bin/p/bin_bflt.c [hunk 1]
```diff
@@ -72,7 +72,7 @@ static RList *patch_relocs(RBin *b) {
 	if (!b || !b->iob.io || !b->iob.io->desc) {
 		return NULL;
 	}
-	if (!b->iob.io->cached) {
+	if (!((b->iob.io->cached >> R_IO_WRITE) & 1)) {
```
Comment (radare) [inline libr/bin/p/bin_bflt.c]: why shift and and, when only & is needed?

Comment (radare) [reply libr/bin/p/bin_bflt.c]: oh wait..

**Comment by radare** [comment]
please resolve conflicts

**Comment by MaskRay** [comment]
Rebased.

Did not notice `R_IO_READ` is shifted.

File: libr/core/canal.c [hunk 2]
```diff
@@ -3645,9 +3645,7 @@ R_API void r_core_anal_esil(RCore *core, const char *str, const char *target) {
 	//this is necessary for the hook to read the id of analop
 	ESIL->user = &op;
 	ESIL->cb.hook_mem_read = &esilbreak_mem_read;
-	if (!core->io->cached) {
-		ESIL->cb.hook_mem_write = &esilbreak_mem_write;
-	}
+	ESIL->cb.hook_mem_write = &esilbreak_mem_write;
```
Comment (condret) [inline libr/core/canal.c]: +1 some ppl will dislike it, but this is how it should be

File: libr/core/cconfig.c [hunk 3]
```diff
@@ -2649,7 +2667,9 @@ R_API int r_core_config_init(RCore *core) {
 	SETCB ("io.buffer", "false", &cb_iobuffer, "Load and use buffer cache if enabled");
 	SETI ("io.buffer.from", 0, "Lower address of buffered cache");
 	SETI ("io.buffer.to", 0, "Higher address of buffered cache");
-	SETCB ("io.cache", "false", &cb_iocache, "Enable cache for io changes");
+	SETCB ("io.cache", "false", &cb_io_cache, "Change both of io.cache.{read,write}");
```
Comment (condret) [inline libr/core/cconfig.c]: was it necessary to change the description? just curios

File: libr/core/canal.c (hunk 2 above)
Comment (MaskRay) [reply libr/core/canal.c]: As a notice `esilbreak_mem_write` is not implemented.

I hate unnecessary checks, generally.

File: libr/core/cconfig.c (hunk 3 above)
Comment (MaskRay) [reply libr/core/cconfig.c]: I changed it b/c I did not find a better description. It is edge-triggered, not level-triggered.

**Pull request merged**